server/commands/music.py

```diff
@@ -343,3 +350,248 @@ def ooc_cmd_random_music(client, arg):
+        if "category" in musiclist[i]:
+            categories.append(musiclist[i]["category"])
+        else:
+            categories.append(None)
+
+    if f"=={args[1]}==" not in categories:
+        client.send_ooc("Category has not been found!")
+        return
+
+    category_id = categories.index(f"=={args[1]}==")
+
+    songs = [
+        musiclist[category_id]["songs"][i]["name"]
+        for i in range(0, len(musiclist[category_id]["songs"]))
+    ]
+
+    if args[2] not in songs:
+        client.send_ooc("Song has not been found!")
+        return
+
+    song_id = songs.index(args[2])
+    musiclist[category_id]["songs"].pop(song_id)
+    if len(musiclist[category_id]["songs"]) == 0:
+        musiclist.pop(category_id)
+
+    if args[0] == "local":
+        path = client.music_ref
+        client.music_list = musiclist_rebuild(musiclist, path)
+    elif args[0] == "area":
+        path = client.area.music_ref
+        client.area.music_list = musiclist_rebuild(musiclist, path)
+    else:
+        path = client.area.area_manager.music_ref
+        client.area.area_manager.music_list = musiclist_rebuild(musiclist, path)
+        
+    client.server.client_manager.refresh_music(targets, True)
+    client.send_ooc(f"'{args[2]}' song has been removed to '{path}' musiclist.")
+
+
+def ooc_cmd_musiclist_add(client, arg):
+    """
+    Allow you to add a song in a loaded musiclist!
+    Remember to insert also Music extension in <MusicName>
+    Usage: /musiclist_add <local/area/hub> <Category> <MusicName>
+    """
+    if arg == "":
+        client.send_ooc("Usage: /musiclist_add <local/area/hub> <Category> <MusicName>")
+        return
+
+    args = shlex.split(arg)
+    if len(args) != 3:
+        client.send_ooc("Usage: /musiclist_add <local/area/hub> <Category> <MusicName>")
+        return
+    
+    if args[0] not in ["local", "area", "hub"]:
+        client.send_ooc("You can add a song if musiclist is loaded in local or in area or in hub.\nUsage: /musiclist_add <local/area/hub> <Category> <MusicName>")
+        return
+    
+    if args[0] == "local":
+        targets = [client]
+        musiclist = client.music_list
+    elif args[0] == "area":
+        if client not in client.area.owners and client not in client.area.area_manager.owners and not client.is_mod:
+            client.send_ooc("You should be at least cm to add a song in a musiclist!")
+            return
+        targets = client.area.clients
+        musiclist = client.area.music_list
+    else:
+        if client not in client.area.area_manager.owners and not client.is_mod:
+            client.send_ooc("You should be at least gm to add a song in a musiclist!")
+            return
+        targets = client.area.area_manager.clients
+        musiclist = client.area.area_manager.music_list
+        
+    if musiclist == []:
+        client.send_ooc("You can add a song if a musiclist is loaded in local or in area or in hub.")
+        return
```

Comment:
the command should not be blocked because a music list is not loaded. After all /save_musiclist exists, use that if you want it saved